internal/util/helpers_test.go

```diff
@@ -4,18 +4,767 @@
+				err := os.WriteFile(tc.filePath, []byte(tc.expectedDesc), 0600)
+				assert.NoError(t, err)
+				defer os.Remove(tc.filePath)
+			}
+
+			desc, closer, err := util.OpenFileArg(tc.filePath, tc.dashOpen)
+			if closer != nil {
+				defer closer()
+			}
+
+			if tc.expectError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				if tc.filePath == "-" || tc.filePath != "-" {
+					buf := new(strings.Builder)
+					_, err := io.Copy(buf, desc)
+					assert.NoError(t, err)
+					assert.Equal(t, tc.expectedDesc, buf.String())
+				}
+			}
+		})
+	}
+}
+
+// TestExpandFileArgs tests the ExpandFileArgs function.
+func TestExpandFileArgs(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name     string
+		files    []string
+		expected []util.ExpandedFile
+		wantErr  bool
+	}{
+		{
+			name:     "Single file",
+			files:    []string{"testfile.txt"},
+			expected: []util.ExpandedFile{{Path: "testfile.txt", Expanded: false}},
+			wantErr:  false,
+		},
+
+		{
+			name:     "Single directory",
+			files:    []string{"testdir"},
+			expected: []util.ExpandedFile{{Path: "testdir/file1.txt", Expanded: true}, {Path: "testdir/file2.txt", Expanded: true}},
+			wantErr:  false,
+		},
+		{
+			name:     "File and directory",
+			files:    []string{"testfile.txt", "testdir"},
+			expected: []util.ExpandedFile{{Path: "testfile.txt", Expanded: false}, {Path: "testdir/file1.txt", Expanded: true}, {Path: "testdir/file2.txt", Expanded: true}},
+			wantErr:  false,
+		},
+		{
+			name:     "File with '-'",
+			files:    []string{"-"},
+			expected: []util.ExpandedFile{{Path: "-", Expanded: false}},
+			wantErr:  false,
+		},
+		{
+			name:     "Non-existent file",
+			files:    []string{"nonexistent.txt"},
+			expected: nil,
+			wantErr:  true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			// Create a temporary directory
+			testDir := t.TempDir()
+			if err := setupTestFiles(testDir); err != nil {
+				t.Fatalf("Failed to set up test files: %v", err)
+			}
+			// Update file paths to include the unique directory
+			for i, file := range tt.files {
+				tt.files[i] = fmt.Sprintf("%s/%s", testDir, file)
+			}
+			for i, file := range tt.expected {
+				tt.expected[i].Path = fmt.Sprintf("%s/%s", testDir, file.Path)
+			}
+			combinedFiles := tt.files
+			got, err := util.ExpandFileArgs(combinedFiles...)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ExpandFileArgs() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			if !compareExpandedFiles(got, tt.expected) {
+				t.Errorf("ExpandFileArgs() = %v, want %v", got, tt.expected)
+			}
+			defer os.RemoveAll(testDir)
+		})
+	}
+}
+
+func compareExpandedFiles(got, expected []util.ExpandedFile) bool {
+	if len(got) != len(expected) {
+		return false
+	}
+	for i := range got {
+		if got[i].Path != expected[i].Path || got[i].Expanded != expected[i].Expanded {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
Consider using [`slices.Equal`](https://pkg.go.dev/slices#Equal), or `slices.EqualFunc` with an anonymous function if you need to.

Reply:
I will use slices.EqualFunc with an anonymous function